### DISCUSSION TITLE
How to get Zim to open a file to the same page every time.

### DISCUSSION THREAD

**Discussion by ghost** [zim-desktop-wiki/zim-desktop-wiki#2027] [Q&A] [answered]
I open a second Zim file from within my journal. That file often fails to open to the last page I was using. Is there a fix for this?

**Comment by introt** [comment, marked as answer]
Hi! Could you elaborate on how you open the second notebook (ie. via the menu, a file/page link, or an interwiki link)? What's your Zim version?

As for a workaround whatever the version, "to open a file to the same page every time" you can link to that page using either an interwiki link or a link to the page's source file - either one should work.

**Reply by ghost** [reply]
So sorry. I should know better. I provided almost no information and even posted my query in the wrong place! Old people shouldn't post late at night after long days!

I'm using Zim version 0.74.3 running under the MUSL version of Void GNU/Linux and XFCE desktop environment.

My default file in Zim is a journal. I open the other files using the Ctrl-O key combination. (I almost never touch a mouse.) Those other files contain something on the order of 100 pages.

Let's say I start Zim first thing in the morning and eventually need to refer to one of these other files. I hit Ctrl-O, and the other file opens to the last page I used within that file. If I now close that second file, and then open in again later in that session, it will open to its home page instead of the page I was just using.

If I open the second file, go to a specific page, close the second file, then close my default Zim window -- the problem does not occur. In other words:

1. Open main Zim window (a journal).
2. Open another Zim file and select the page I need to see.
3. Close the other Zim file.
4. Re-open the other Zim file.
5. The displayed page will be the home page of the other Zim file instead of the last page I used.

To eliminate the problem:

1. Open main Zim window (a journal).
2. Open another Zim file and select the page I need to see.
3. Close the other Zim file.
4. Close the main Zim window.
5. Re-open the main Zim window.
6. Re-open the other Zim file.
7. The displayed page will be the correct page.

So the state of the second window isn't preserved within the same Zim session, but it is preserved if I quit Zim entirely.

I suppose this might actually be a minor bug, but I certainly didn't report it properly on my first attempt. As I said, I was pretty spacey  last night!

Thanks!

**Reply by ghost** [reply]
Hi, thanks for the suggestion. I'll see if I can make this idea work, but in a given session I may use different pages rather than always the same page within that file. I guess it's obvious that I could just not close the page in the first place. Actually, your suggestion might work better overall for my purposes. I'll try it out to see.

In the meantime, is this behavior that I've reported an intended behavior, or just unforeseen? I don't remember seeing this issue in Zim over the years, but I may have just become inured to it and only noticed it after a change in work flow.

**Reply by ghost** [reply]
I adjusted my work flow slightly to use your suggestion of using an inter-wiki link. This is just fine.

I am wondering if I should report the aforementioned behavior as a bug. Seems like the file's state should be saved when the file is closed rather than when Zim is quit completely. Do you have an opinion on that?

Thanks much for your suggestion. It works better than opening the file and then using Ctrl-J to jump to the page I want.

**Reply by introt** [reply]
No worries, apologies for the late answer.

I assume you're referring to separate notebooks as "files" - while each page is its own file, each notebook does have its own `notebook.zim` configuration file. ~These files should have the line `shared=False` to keep their respective state - see eg. #2019 for more information.~ nvm, that doesn't solve it.

Other state (windows size etc) does work as it should, so that's definitely a bug; can be reproduced with your thorough steps, thanks for taking the time to write them down! For the time being, using the `--standalone` flag (combined with `shared=False`?) seems to work around the issue.

**Reply by ghost** [reply]
Yes, that's correct. I was referring to separate notebooks as files, though I know that they are actually collections of files. I was thinking of the way the open notebook dialog works, more or less the same way most applications open individual files.

Thank you for confirming that the behavior is a bug. Since you're a collaborator should I consider the bug reported?

I really appreciate your help. I'm grateful to all of you for developing and maintaining what has turned out to be my single most useful application!

**Reply by introt** [reply]
Sure, I can take it over from here - given your thorough description, I can simply copy-paste the relevant bits into the right template - just need to check whether the issue is present on the `develop` branch, and whether it's easy enough for me to fix in order to jump straight to a PR.

On behalf of all the contributors, you're very welcome, and thanks to you too for reporting issues!
